java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -333,22 +337,32 @@ public BufferAllocator newChildAllocator(
       final String name,
       final long initReservation,
       final long maxAllocation) {
+    return newChildAllocator(name, this.listener, initReservation, maxAllocation);
+  }
+
+  @Override
+  public BufferAllocator newChildAllocator(
+      final String name,
+      final AllocationListener listener,
+      final long initReservation,
+      final long maxAllocation) {
     assertOpen();
 
-    final ChildAllocator childAllocator = new ChildAllocator(this, name, initReservation,
-        maxAllocation);
+    final ChildAllocator childAllocator = new ChildAllocator(listener,this, name, initReservation,
+                                                             maxAllocation);
```

Comment:
weird indenting here and elsewhere.

Reply:
Fixed. Please note that this passed the check style